sql/2021/security/feature_adoption_by_technology.sql

```diff
@@ -0,0 +1,43 @@
+#standardSQL
+# Adoption of features based on the technology that is used
+SELECT
+  client,
+  category,
+  app,
+  headername,
+  COUNT(DISTINCT url) AS total_pages_with_technology,
+  COUNT(DISTINCT IF(STARTS_WITH(url, 'https'), url, NULL)) AS total_https_pages,
+  COUNT(DISTINCT IF(REGEXP_CONTAINS(respOtherHeaders, CONCAT('(?i)', headername, ' ')), url, NULL)) AS freq,
+  SAFE_DIVIDE(COUNT(DISTINCT IF(REGEXP_CONTAINS(respOtherHeaders, CONCAT('(?i)', headername, ' ')), url, NULL)), COUNT(DISTINCT url)) AS pct,
+  SAFE_DIVIDE(COUNT(DISTINCT IF(REGEXP_CONTAINS(respOtherHeaders, CONCAT('(?i)', headername, ' ')) AND STARTS_WITH(url, 'https'), url, NULL)), COUNT(DISTINCT IF(STARTS_WITH(url, 'https'), url, NULL))) AS pct_https
```

Comment:
Not convinced about these totals due to UNNEST and HAVING clauses. When I run the simple counts on technologies tables I get different totals.

I think this needs to be a CTE or a subquery.

See: https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Analysts'-Guide#a-warning-on-totals